aten/src/ATen/native/Convolution.cpp

```diff
@@ -395,7 +395,10 @@ at::Tensor _convolution_nogroup(
     if (dim == 4) {
       if (dilated) {
         return at::thnn_conv_dilated2d(
-            input, weight, kernel_size, bias,
+            // TODO: This might not be necessary in the non-CUDA case,
+            // as we only triggered a failure here with
+            // test_ConvTranspose1d_cuda.  Very unsatisfactory...
+            input, weight.contiguous(), kernel_size, bias,
```

Comment:
This should be fixed in TH(CU)NN itself (add `newContiguous` for `weight`)

Reply:
So, I took a look at THCUNN and actually, THCUNN is already running `newContiguous` for weight, and the check was added by you in 45596d52897fb187701943cb77456ff1e7249989

I am going to delete this test :)